File: GXpert/GXpert.Web/Modules/Masters/Project/ProjectColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace GXpert.Masters.Columns;

[ColumnsScript("Masters.Project")]
[BasedOnRow(typeof(ProjectRow), CheckNames = true)]
public class ProjectColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    public string StudentPrn { get; set; }
    public string CourseTitle { get; set; }
    public string ClassTitle { get; set; }
    public string SemesterTitle { get; set; }
    [EditLink]
    public string Title { get; set; }
    public string Description { get; set; }
    public short EType { get; set; }
}

File: GXpert/GXpert.Web/Modules/ServerTypes/Masters/ProjectColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { EProjectType } from "../Web/Enums.EProjectType";
import { ProjectRow } from "./ProjectRow";

export interface ProjectColumns {
    Id: Column<ProjectRow>;
    StudentPrn: Column<ProjectRow>;
    CourseTitle: Column<ProjectRow>;
    ClassTitle: Column<ProjectRow>;
    SemesterTitle: Column<ProjectRow>;
    Title: Column<ProjectRow>;
    Description: Column<ProjectRow>;
    EType: Column<ProjectRow>;
}

export class ProjectColumns extends ColumnsBase<ProjectRow> {
    static readonly columnsKey = 'Masters.Project';
    static readonly Fields = fieldsProxy<ProjectColumns>();
}

[EProjectType]; // referenced types